### ISSUE TITLE
About the new 1.3.4 attachment items system

### ISSUE THREAD

**Issue by BrownBear85** [Lightning-64/Tide#36]
Thanks for implementing the ability to add custom bobber items! I appreciate the quick response. Stardew Fishing is now compatible with Tide!

Firstly, I'd just like to say that obviously, this is your mod and you can make it however you want. I don't want to be the annoying guy who tells you what to do. But in my opinion, Aquaculture's fishing rod attachment system is the most balanced and works the best for Stardew Fishing (it's also how bobbers in Stardew Valley, the game our minigame is based off of, works). In that system, bobber items are consumed when applied to fishing rods, but can be retrieved by removing them from the fishing rod (essentially you are storing the bobber item inside of the fishing rod item). Such a system was what I was suggesting in my earlier issue, and I'd just like to make sure you don't change your mod based on a misinterpretation of my suggestion.

Your current system of being able to essentially copy a single bobber onto multiple fishing rods conflicts with the vision we have for Stardew Fishing's bobbers, since they have a large gameplay impact beyond cosmetics and even have durability and can break. However, like I said, I'm talking about *our* vision for *our* bobbers, so if you have a different vision for your bobbers, then obviously it's your choice which system you go with, and that's fine.

That being said, It would be great if you could consider tweaking your attachment system to store attachment items in nbt as itemstacks and adopt my proposed system of consuming and retrieving, something that I believe to be more balanced for both our mods.

Thanks again for your helpfulness!

P.S. It looks like tooltips don't properly show the name of custom bobber items: (this is with one of my bobber items equipped)
![image](https://github.com/user-attachments/assets/1b4c386f-d4e6-4e13-aaad-04b3fea416b9)

**Comment by Lightning-64** [comment]
So first of all, for the tooltips, did you pass in a custom translation key for the bobber tooltip? It uses a different translation key separate from the base item name.

Secondly, I have no problem with changing the system a bit; I think it's great that I can get feedback from other people so I can make the mod how they want it to be played, and I have no problem doing it again! I didn't realize initially that you were talking about storing the items in the rod itself and getting them back in the same way. I'll try to add this to the next update when I get the chance!

Sorry for all the confusion :/

**Comment by BrownBear85** [comment]
Awesome, thanks! No worries about the confusion, that was probably my fault. And thanks for letting me know about the need for a bobber translation key. I should've done some more digging before I assumed it was a big!

**Comment by Lightning-64** [comment]
Ok, so I posted a [commit](https://github.com/Lightning-64/Tide/commit/0be30de5b5375c2d02db22f56a575dd267731578) in the 1.20.1 branch (that will soon be added to the 1.21.1 branch as well) that kind of changes the system in which accessory data is passed to the fishing rod. Instead of storing the texture location and rod translation through nbt, it now checks the item class of the bobber/hook/line and runs the methods `getTextureLocation()` and `getTranslation()`, which return a resourcelocation and a component respectively.

In short, instead of adding to the nbt data of the items, just create those two methods in your custom bobber class that returns the correct values that will be obtained by the rod!

In other news, I was able to get ItemStack support for the rod accessories, so you should be able to change the durability of the rods while they're attached to the fishing rod. Additionally, I slightly reworked the Angler's Workshop to handle keeping the items stored in the rod nbt and retrieving them from the rod like you asked for!

**Comment by BrownBear85** [comment]
That all sounds great! :) I'll take a look soon and let you know if I have any feedback.

**Comment by Lightning-64** [comment]
Tide 1.4 is now released, and it includes the new system that I mentioned! It also properly stores the fishing rod accessories inside of the item stack, so Tide should be good to go for Stardew Fishing compat!
